hamcrest/Hamcrest/Type/IsNumeric.php

```diff
@@ -19,10 +19,36 @@ public function __construct()
 
     public function matches($item)
     {
+        if ($this->is_hexadecimal($item)) {
+
+            return true;
+        }
+
         return is_numeric($item);
     }
 
     /**
+     * This check is necessary because PHP 7 doesn't recognize hexadecimal string as numeric anymore.
+     *
+     * If $item is a string and that the string is long enough to be hexadecimal
+     * Check if it's hexadecimal
+     * And if it is, decode it before to return it to is_numeric.
+     *
+     * @param mixed $item
+     * @return boolean
+     */
+    private function is_hexadecimal($item)
+    {
+        if (is_string($item) && substr($item, 0, 2) === '0x' ) {
```

Comment:
Maybe we can improve this even more by:

* doing `preg_match` to ensure, that:
 * string starts with `0x`
 * has something else after `0x`
 * capturing that

This way we can use capture result right inside `if` statement without doing extra `substr`.